src/hb-open-type.hh

```diff
@@ -338,6 +338,30 @@ struct OffsetTo : Offset<OffsetType, has_null>
   }
 
   /* TODO: Somehow merge this with previous function into a serialize_dispatch(). */
+  /* Workaround clang bug: https://bugs.llvm.org/show_bug.cgi?id=23029
+   * Can't compile: whence = hb_serialize_context_t::Head followed by Ts&&...
+   */
+  template <typename ...Ts>
+  bool serialize_copy (hb_serialize_context_t *c,
+		       const OffsetTo& src,
+		       const void *src_base,
+		       const void *dst_base,
+		       hb_serialize_context_t::whence_t whence,
+		       Ts&&... ds)
```

Comment:
This is a problem now since next function doesn't have whence but still has parameter pack!  Correct fix is for next function to have only defaulted whence and no parameter pack.

Reply:
Oh, and obviously one of them should call the other.  Why would you ever duplicate code like that?!

Reply:
Do we actually need this flavor?  It already takes src AND dst base.  Why does it need whence?!

Reply:
AFAICT, second one is now used without parameter pack so it can be removed. And duplicating code is certainly not clean. I can change second to call first.

Reply:
Second serialize_copy with defaulting whence without parameter pack chaining to first won't work, because second calls itself recursively.
```
  bool serialize_copy (hb_serialize_context_t *c,
		       const OffsetTo& src,
		       const void *src_base,
		       const void *dst_base,
		       hb_serialize_context_t::whence_t whence = hb_serialize_context_t::Head)
  { return serialize_copy (c, src, src_base, dst_base, whence); }
```
 I think I will remove whence as well, and specify Head as whence when chaining to first.

Reply:
Right.  Make second not take whence at all and implicitly default.

Reply:
Or whichever way you find.  Am happy as long  as not duplicated code.  Thanks!

Reply:
Modified second definition of serialize_copy() to call first as I described above.
I believe this PR is ready to merge.